pkg/isoeditor/nmstate_handler.go

```diff
@@ -0,0 +1,127 @@
+package isoeditor
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+//go:generate mockgen -package=isoeditor -destination=mock_nmstate_handler.go . NmstateHandler
+type NmstateHandler interface {
+	CreateNmstateRamDisk(rootfsPath, ramDiskPath string) error
+}
+
+type nmstateHandler struct {
+	workDir  string
+	executer Executer
+}
+
+func NewNmstateHandler(workDir string, executer Executer) NmstateHandler {
+	return &nmstateHandler{
+		workDir:  workDir,
+		executer: executer,
+	}
+}
+
+func (n *nmstateHandler) CreateNmstateRamDisk(rootfsPath, ramDiskPath string) error {
+	// Extract nmstatectl binary
+	var err error
+	nmstateDir := filepath.Join(n.workDir, "nmstate")
+	err = os.MkdirAll(nmstateDir, os.ModePerm)
+	if err != nil {
+		return err
+	}
+	binaryPath, err := n.extractNmstatectl(rootfsPath, nmstateDir)
+	if err != nil {
+		return err
+	}
+	nmstatectlPath := filepath.Join(nmstateDir, "squashfs-root", binaryPath)
+
+	// Check if nmstatectl binary file exists
+	if _, err = os.Stat(nmstatectlPath); os.IsNotExist(err) {
+		return err
+	}
+
+	// Read binary
+	nmstateBinContent, err := os.ReadFile(nmstatectlPath)
+	if err != nil {
+		return err
+	}
+
+	// Create a compressed RAM disk image with the nmstatectl binary
+	compressedCpio, err := generateCompressedCPIO(nmstateBinContent, NmstatectlPathInRamdisk, 0o100_755)
+	if err != nil {
+		return err
+	}
+
+	// Write RAM disk file
+	err = os.WriteFile(ramDiskPath, compressedCpio, 0755) //nolint:gosec
+	if err != nil {
+		return err
+	}
+
+	// Remove temp dir
+	if err = os.RemoveAll(nmstateDir); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (n *nmstateHandler) extractNmstatectl(rootfsPath, nmstateDir string) (string, error) {
+	_, err := n.executer.Execute(fmt.Sprintf("7z x %s", rootfsPath), nmstateDir)
+	if err != nil {
+		log.Errorf("failed to 7z x rootfs.img: %v", err.Error())
+		return "", err
+	}
+	// limiting files is needed on el<=9 due to https://github.com/plougher/squashfs-tools/issues/125
+	ulimit := "ulimit -n 1024"
+	list, err := n.executer.Execute(fmt.Sprintf("%s ; unsquashfs -d '' -lc %s", ulimit, "root.squashfs"), nmstateDir)
```

Comment:
What is `-lc`? I don't see a `-lc` or `-c` option in `man unsquashfs` on my system.

Also does  `-d ''` unpack into the working directory?
Maybe just add a comment explaining the `unsquashfs` options you're using here.

Reply:
Also maybe use the non-abbreviated version on the options like you do on the later command.